pkgs/dart_model/lib/src/json_buffer/json_buffer_builder.dart

```diff
@@ -297,17 +308,27 @@ class JsonBufferBuilder {
     _buffer.setRange(0, oldBuffer.length, oldBuffer);
   }
 
+  /// Throws if writes are not allowed.
   void _checkAllowWrites() {
     if (!_allowWrites) throw StateError('This JsonBufferBuilder is read-only.');
   }
 
+  /// Explanations of each byte in the buffer.
+  ///
+  /// Run tests with `dart -Ddebug_json_buffer=true test -c source` to enable.
+  ///
+  /// Then, [JsonBufferBuilder#toString] prints additional information for
+  /// each byte.
+  final _Explanations? _explanations =
+      const bool.fromEnvironment('debug_json_buffer') ? _Explanations() : null;
```

Comment:
I was originally thinking system environment variables not -D ones, but I see now that the former can't be const.

We could file an issue on package:test to support these, fwiw. Not sure we would actually implement it, there could be some pushback, but it might be worth thinking about.

Reply:
Thanks, filed https://github.com/dart-lang/test/issues/2272